app/screens/image_preview/image_preview.test.js

```diff
@@ -108,4 +115,111 @@ describe('ImagePreview', () => {
+        const files = [{
+            caption: 'caption',
+            data: {
+                mime_type: 'video/mp4',
+            },
+        }];
+        const props = {
+            ...baseProps,
+            index,
+            files,
+        };
+        const wrapper = shallow(
+            <ImagePreview
+                {...props}
+            />,
+            {context: {intl: {formatMessage}}},
+        );
+
+        const instance = wrapper.instance();
+        await instance.showDownloadOptionsIOS();
+
+        const expectedOptions = {
+            options: ['Save Video', 'Cancel'],
+            cancelButtonIndex: 1,
+            anchor: null,
+            title: files[index].caption,
+        };
+        expect(BottomSheet.showBottomSheetWithOptions).
+            toHaveBeenCalledWith(expectedOptions, expect.any(Function));
+    });
+
+    test('should not show bottom sheet when showDownloadOptionsIOS is called for non-existing video', async () => {
+        BottomSheet.showBottomSheetWithOptions = jest.fn();
+        RNFetchBlob.fs.exists = jest.fn().mockReturnValue(false);
+
+        const index = 0;
+        const files = [{
+            caption: 'caption',
+            data: {
+                mime_type: 'video/mp4',
+            },
+        }];
+        const props = {
+            ...baseProps,
+            index,
+            files,
+        };
+        const wrapper = shallow(
+            <ImagePreview
+                {...props}
+            />,
+            {context: {intl: {formatMessage: jest.fn()}}},
+        );
+
+        const instance = wrapper.instance();
+        await instance.showDownloadOptionsIOS();
+
+        expect(BottomSheet.showBottomSheetWithOptions).not.toHaveBeenCalled();
+    });
+
+    test('should show bottom sheet when showDownloadOptionsIOS is called for image', async () => {
+        BottomSheet.showBottomSheetWithOptions = jest.fn();
+        RNFetchBlob.fs.exists = jest.fn().mockReturnValue(true);
+        const formatMessage = jest.fn(({defaultMessage}) => {
+            return defaultMessage;
+        });
+
+        const index = 0;
+        const files = [{
+            caption: 'caption',
+            data: {
+                mime_type: 'image/jpeg',
+            },
+        }];
+        const props = {
+            ...baseProps,
+            index,
+            files,
+        };
+        const wrapper = shallow(
+            <ImagePreview
+                {...props}
+            />,
+            {context: {intl: {formatMessage}}},
+        );
+
+        const instance = wrapper.instance();
+        await instance.showDownloadOptionsIOS();
+
+        const expectedOptions = {
+            options: ['Save Image', 'Cancel'],
+            cancelButtonIndex: 1,
+            anchor: null,
+            title: files[index].caption,
+        };
+        expect(BottomSheet.showBottomSheetWithOptions).
+            toHaveBeenCalledWith(expectedOptions, expect.any(Function));
+    });
```

Comment:
💯

Reply:
@saturninoabril single images can be downloaded, open graph images cannot!